### PR TITLE
Cmr 6783 - Adding the horizontal data resolution range facets. This includes indexing and searching.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj [hunk 1]
```diff
@@ -94,24 +97,47 @@
...
+  (if (= field-name :horizontal-data-resolution-range)
+    (let [aggs (get-in aggregations [field-name :values])
+          some-values? (->> aggs
+                            :buckets
+                            (map :doc_count)
+                            (some #(< 0 %)))]
+      (if some-values?
+        (frf/buckets->value-count-pairs
+          (get-in aggregations [field-name :values]))
+        (sequence nil)))
```
Comment (mschmele) [inline search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj:115]: This feels like it should be its own function

File: search-app/src/cmr/search/services/humanizers/humanizer_range_facet_service.clj [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+(defn store-range-facets
+  "Stores the passed in range facets into the range facet cache. All searches will be using range
+   facets initially and caching range facets prevents every inital search call from parsing the
+   range facet humanizers, which keeps the search service fast."
+  [context range-facets]
+  (when range-facets
+    (let [cache (cache/context->cache context range-facet-cache-key)]
+      (when cache
+        (cache/set-value cache range-facet-data-key range-facets)))))
```
Comment (mschmele) [inline search-app/src/cmr/search/services/humanizers/humanizer_range_facet_service.clj:106]: Could you use a `when-let` here?

**Review by mschmele — COMMENTED** [review]
Looks pretty good to me, but this is a big enough change that we should probably review this over a goto so that we all have the proper context

File: common-lib/src/cmr/common/util.clj [hunk 3]
```diff
@@ -996,3 +996,36 @@
...
+;; It is used to convert
+(def decimal-degress->meters
+  "We are using the great circle algorithm https://en.wikipedia.org/wiki/Great-circle_distance
+   as the conversion from Decimal Degrees to meters. (2*pi*r)/360 where r=6371009 meters - the mean
+   radius of the earth using the WGS84 ellipsoid. This conversion is adequate for horizontal data
+   resolution range facets for which this conversion is being used. 2*3.14159*6371009/360=111195
+   meters/degree"
+  111195)
+
+(def kilometers->meters
```
[the remark this answers is not in the document]
Comment (eereiter) [reply common-lib/src/cmr/common/util.clj:1010]: how so?  what do you mean?

File: indexer-app/src/cmr/indexer/data/concepts/collection/resolution.clj [hunk 4]
```diff
@@ -0,0 +1,66 @@
+(ns cmr.indexer.data.concepts.collection.resolution
+  "Contains functions to parse and convert collection resolution information so that
+   the information can be indexed."
+  (:require
+   [cmr.common.util :refer [remove-nils-empty-maps-seqs convert-to-meters]]))
```
[the remark this answers is not in the document]
Comment (eereiter) [reply indexer-app/src/cmr/indexer/data/concepts/collection/resolution.clj:5]: I see the use of refer in the CMR code usually for tests and remove-nil's. I use it usually when I only use 1 or 2 functions in a different name space.  I can change it to as if it is more desirable.

File: common-lib/test/cmr/common/test/util.clj [hunk 5]
```diff
@@ -829,3 +829,46 @@
                     {:c "description2"}
                     {:c "description3"}]}}
            (util/update-in-each-vector update-in-each-vector-test-data [:a :b] dissoc :d)))))
+
+(deftest convert-resolution-to-meters-test
```
[the remark this answers is not in the document]
Comment (eereiter) [reply common-lib/test/cmr/common/test/util.clj:833]: I have added the case when the value is zero.

File: search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj (hunk 1 above)
Comment (eereiter) [reply search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj:115]: This code is in its own function

Comment (eereiter) [reply search-app/src/cmr/search/services/query_execution/facets/facets_v2_results_feature.clj:115]: Then the current function will just be an if statement.

File: search-app/src/cmr/search/services/humanizers/humanizer_range_facet_service.clj (hunk 2 above)
Comment (eereiter) [reply search-app/src/cmr/search/services/humanizers/humanizer_range_facet_service.clj:106]: yes, I have updated the code.

File: common-lib/src/cmr/common/util.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (mschmele) [reply common-lib/src/cmr/common/util.clj:1010]: Oh just name it something like `metric-conversion-factor` or something. The logic wouldn't change, just the name of this. It's a nitpick, so feel free to ignore

**Review by mschmele — APPROVED** [review]

File: common-lib/src/cmr/common/util.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (eereiter) [reply common-lib/src/cmr/common/util.clj]: I have updated the names.

**Review by jaybarra — APPROVED** [review]

**Review by ygliuvt — APPROVED** [review]
update the api doc mentioning the data resolution range facets.

**Pull request merged**